core/src/main/java/io/questdb/cairo/wal/WalMetrics.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.wal;
+
+import io.questdb.metrics.Counter;
+import io.questdb.metrics.DoubleGauge;
+import io.questdb.metrics.LongGauge;
+import io.questdb.metrics.MetricsRegistry;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class WalMetrics {
+    private final DoubleGauge physicallyWrittenRowsMultiplierGauge;
+    private final LongGauge rowsWriteRateGauge;
+    private final Counter rowsWrittenCounter;
+    private final AtomicLong totalRowsPhysicallyWritten;
+    private final AtomicLong totalRowsWritten;
+    private final AtomicLong totalRowsWrittenTotalTime;
+
+    public WalMetrics(MetricsRegistry metricsRegistry) {
+        this.physicallyWrittenRowsMultiplierGauge = metricsRegistry.newDoubleGauge("wal_apply_physically_written_rows_multiplier");
+        this.rowsWrittenCounter = metricsRegistry.newCounter("wal_apply_written_rows");
+        this.rowsWriteRateGauge = metricsRegistry.newLongGauge("wal_apply_rows_per_second");
```

Comment:
This and `wal_apply_physically_written_rows_multiplier` metrics don't add a lot of value as it's trivial to configure Prometheus to calculate them. It's enough to expose two counters: logically and physically written rows. Additionally, we could expose a gauge for the WAL application call latency, but that's not a must-have.

Reply:
Can you show me an example how to configure division of 2 metrics in Prometheus please? I was looking for a way but didn't find it

Reply:
I was referring to Grafana, not Prometheus. Sorry for the confusion. In Grafana it's trivial to write a formula when creating a new chart.

Reply:
Ok, I changed to expose `wal_apply_physically_written_rows` instead of multiplier.
Latencies are calculated in https://github.com/questdb/questdb/pull/2930 we can add it as metrics when these 2 merged.

Reply:
Probably we don't need `wal_apply_rows_per_second` as it's trivial to configure this in Grafana.

Ideally, we also need written rows metric in `WalWriter`s, but it's better to add it in a separate PR.

Reply:
> Ideally, we also need written rows metric in `WalWriter`s, but it's better to add it in a separate PR.

I'll submit a separate PR for that.